pkg/controller/controller_binding_test.go

```diff
@@ -3788,6 +3801,195 @@ func TestPollServiceBinding(t *testing.T) {
+				{
+					AddKey: &v1beta1.AddKeyTransform{
+						Key:                "bar",
+						JSONPathExpression: pointer("{.foo}"),
+					},
+				},
+			},
+			credentials: map[string]interface{}{
+				"foo": "123",
+			},
+			transformedCredentials: map[string]interface{}{
+				"foo": "123",
+				"bar": "123",
+			},
+		},
+		{
+			name: "AddKeyTransform with JSONPathExpression on non-flat credentials",
+			transforms: []v1beta1.SecretTransform{
+				{
+					AddKey: &v1beta1.AddKeyTransform{
+						Key:                "child-of-foo",
+						JSONPathExpression: pointer("{.foo.child}"),
+					},
+				},
+			},
+			credentials: map[string]interface{}{
+				"foo": map[string]interface{}{
+					"child": "123",
+				},
+			},
+			transformedCredentials: map[string]interface{}{
+				"foo": map[string]interface{}{
+					"child": "123",
+				},
+				"child-of-foo": "123",
+			},
+		},
+		{
+			name: "AddKeyTransform stringValue precedence over value",
+			transforms: []v1beta1.SecretTransform{
+				{
+					AddKey: &v1beta1.AddKeyTransform{
+						Key:         "bar",
+						Value:       []byte("456"),
+						StringValue: pointer("789"),
+					},
+				},
+			},
+			credentials: map[string]interface{}{
+				"foo": "123",
+			},
+			transformedCredentials: map[string]interface{}{
+				"foo": "123",
+				"bar": "789",
+			},
+		},
+		{
+			name: "MergeSecretTransform",
+			transforms: []v1beta1.SecretTransform{
+				{
+					AddKeysFrom: &v1beta1.AddKeysFromTransform{
+						SecretRef: &v1beta1.ObjectReference{
+							Namespace: "ns",
+							Name:      "other-secret",
+						},
+					},
+				},
+			},
+			credentials: map[string]interface{}{
+				"foo": []byte("123"),
+			},
+			otherSecret: &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "ns",
+					Name:      "other-secret",
+				},
+				Data: map[string][]byte{
+					"bar": []byte("456"),
+				},
+			},
+			transformedCredentials: map[string]interface{}{
+				"foo": []byte("123"),
+				"bar": []byte("456"),
+			},
+		},
+		{
+			name: "RemoveKeyTransform",
+			transforms: []v1beta1.SecretTransform{
+				{
+					RemoveKey: &v1beta1.RemoveKeyTransform{
+						Key: "bar",
+					},
+				},
+			},
+			credentials: map[string]interface{}{
+				"foo": "123",
+				"bar": "456",
+			},
+			transformedCredentials: map[string]interface{}{
+				"foo": "123",
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		fakeKubeClient, _, _, testController, _ := newTestController(t, fakeosb.FakeClientConfiguration{})
+
+		if tc.otherSecret != nil {
+			addGetSecretReaction(fakeKubeClient, tc.otherSecret)
+		}
+
+		err := testController.transformCredentials(tc.transforms, tc.credentials)
+		if err != nil {
+			t.Fatalf("unexpected error: %v", err)
+		}
+		if !reflect.DeepEqual(tc.credentials, tc.transformedCredentials) {
+			t.Errorf("%v: unexpected transformed secret data; expected: %v; actual: %v", tc.name, tc.transformedCredentials, tc.credentials)
+		}
+	}
+}
+
+func pointer(s string) *string {
```

Comment:
we have this function other places and it is called [strPtr](https://github.com/kubernetes-incubator/service-catalog/search?utf8=%E2%9C%93&q=strPtr&type=)

Reply:
Thanks! Fixed now.

Reply:
Should this function be removed now?